llvm/lib/Target/AMDGPU/SIPeepholeSDWA.cpp

```diff
@@ -1285,10 +1374,12 @@ bool SIPeepholeSDWA::run(MachineFunction &MF) {
 
       for (const auto &OperandPair : SDWAOperands) {
         const auto &Operand = OperandPair.second;
-        MachineInstr *PotentialMI = Operand->potentialToConvert(TII, ST, &PotentialMatches);
-        if (PotentialMI && isConvertibleToSDWA(*PotentialMI, ST, TII)) {
+        MachineInstr *PotentialMI =
+            Operand->potentialToConvert(TII, ST, &PotentialMatches);
+
+        if (PotentialMI && isConvertibleToSDWA(*PotentialMI, ST, TII) &&
+            Operand->canCombineSelections(*PotentialMI, TII))
```

Comment:
Think this is redundant if we're already checking from `Operand->potentialToConvert` above

Reply:
Before my last commit, there was no equivalent check in `potentialToConvert` (i.e. on the return value of that function), but I have now moved the call to `canCombineSelections` there; see also https://github.com/llvm/llvm-project/pull/123221#discussion_r1967148059. The call that already existed in `SDWASrcOperand::potentialToConvert` checks that the uses of the operand can also be converted to SDWA instructions. This is necessary in addition to the check of the return value of `potentialToConvert` since the uses are directly added to `PotentialMatches` from there (see function `op_select_byte_0_instr_select_word_1` in `llvm/test/CodeGen/AMDGPU/sdwa-peephole-instr-combine-sel2.mir` for an example which leads to a crash if this check is removed).